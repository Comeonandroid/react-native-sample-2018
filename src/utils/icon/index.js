import { createIconSetFromIcoMoon } from 'react-native-vector-icons'
import styled from 'styled-components'

import icoMoonConfig from './selection.json'
const CustomIcon = createIconSetFromIcoMoon(icoMoonConfig)

export default styled(CustomIcon) `
  color: ${ ({ theme, color }) => color || theme.colors.border };
`

// IMPORTANT
// If you want to add/remove icons please read this first
// https://www.reactnative.guide/12-svg-icons-using-react-native-vector-icons/12.1-creating-custom-iconset.html