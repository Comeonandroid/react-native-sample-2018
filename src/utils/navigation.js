import React from 'react'

import { t } from '../locales/i18n'

import HeaderMenuIcon from '../modules/navigation/header/components/HeaderMenuIcon'

export const getNavigationHeader = ({ navigation, screenProps }) => (
  {
    headerMode: 'float',
    title: t('route.' + navigation.state.routeName),
    headerTintColor: screenProps.theme.colors.headerContrastColor,
    headerStyle: {
      backgroundColor: screenProps.theme.colors.background,
      borderBottomColor: 'transparent',
    },
    headerRight: <HeaderMenuIcon navigation={navigation} />,
    // TODO: header title for all the pages
    headerBackTitle: null
  }
)

export const getHomeHeader = ({ navigation, screenProps }) => (
  {
    headerTintColor: screenProps.theme.colors.headerContrastColor,
    headerStyle: {
      backgroundColor: screenProps.theme.colors.background,
      borderBottomColor: 'transparent',
    },
    headerRight: <HeaderMenuIcon navigation={navigation} />
  }
)
