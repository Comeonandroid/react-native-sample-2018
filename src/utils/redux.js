import {
  createReactNavigationReduxMiddleware,
  createReduxBoundAddListener,
} from 'react-navigation-redux-helpers'

const reduxRouterMiddleware = createReactNavigationReduxMiddleware(
  "root",
  state => state.navigation,
)
const addListener = createReduxBoundAddListener('root')

export {
  reduxRouterMiddleware,
  addListener,
}
