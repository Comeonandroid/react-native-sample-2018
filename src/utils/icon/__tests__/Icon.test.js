import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'

import { itBehavesLike } from '../../../specs/shared_examples/index'
import { wrapWithTheme } from '../../../specs/utils'

import selection from '../selection.json'
import Icon from '../index'

const setup = propOverrides => {
  const props = Object.assign({
    name: 'password',
    size: 25,
    color: null
  }, propOverrides)

  const component = <Icon {...props} />
  const tree = renderer.create(wrapWithTheme(component)).toJSON()

  return {
    props,
    tree,
    component
  }
}

describe('<Icon />', () => {
  const { component } = setup()
  itBehavesLike('aThemedComponent', { subject: component })

  describe('size', () => {
    const { component } = setup({ size: 18 })
    itBehavesLike('aThemedComponent', { subject: component })
  })

  describe('name', () => {
    const { component } = setup({ name: 'cancel' })
    itBehavesLike('aThemedComponent', { subject: component })
  })

  describe('color', () => {
    const color = '#000000'
    const { tree } = setup({ color })
    it('renders passed color in icon styles', () => {
      expect(tree.props.style[1][0].color).toBe(color)
    })
  })

  selection.icons.forEach((icon) => {
    describe(`icon ${icon.properties.name} is presented and renders correctly`, () => {
      const { component } = setup({ name: icon.properties.name })
      itBehavesLike('aThemedComponent', { subject: component })
    })
  })
})
