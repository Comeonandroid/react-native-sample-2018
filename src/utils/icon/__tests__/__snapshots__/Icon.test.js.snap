// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> dark: renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": null,
        "fontSize": 25,
      },
      Array [
        Object {
          "color": "#fcfcfc",
        },
        undefined,
      ],
      Object {
        "fontFamily": "icomoon",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`<Icon /> icon arrow-up is presented and renders correctly dark: renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": null,
        "fontSize": 25,
      },
      Array [
        Object {
          "color": "#fcfcfc",
        },
        undefined,
      ],
      Object {
        "fontFamily": "icomoon",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`<Icon /> icon arrow-up is presented and renders correctly light: renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": null,
        "fontSize": 25,
      },
      Array [
        Object {
          "color": "#4b4c4d",
        },
        undefined,
      ],
      Object {
        "fontFamily": "icomoon",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`<Icon /> icon back is presented and renders correctly dark: renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": null,
        "fontSize": 25,
      },
      Array [
        Object {
          "color": "#fcfcfc",
        },
        undefined,
      ],
      Object {
        "fontFamily": "icomoon",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`<Icon /> icon back is presented and renders correctly light: renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": null,
        "fontSize": 25,
      },
      Array [
        Object {
          "color": "#4b4c4d",
        },
        undefined,
      ],
      Object {
        "fontFamily": "icomoon",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`<Icon /> icon backup is presented and renders correctly dark: renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": null,
        "fontSize": 25,
      },
      Array [
        Object {
          "color": "#fcfcfc",
        },
        undefined,
      ],
      Object {
        "fontFamily": "icomoon",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`<Icon /> icon backup is presented and renders correctly light: renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": null,
        "fontSize": 25,
      },
      Array [
        Object {
          "color": "#4b4c4d",
        },
        undefined,
      ],
      Object {
        "fontFamily": "icomoon",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`<Icon /> icon cancel is presented and renders correctly dark: renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": null,
        "fontSize": 25,
      },
      Array [
        Object {
          "color": "#fcfcfc",
        },
        undefined,
      ],
      Object {
        "fontFamily": "icomoon",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`<Icon /> icon cancel is presented and renders correctly light: renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": null,
        "fontSize": 25,
      },
      Array [
        Object {
          "color": "#4b4c4d",
        },
        undefined,
      ],
      Object {
        "fontFamily": "icomoon",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`<Icon /> icon coin-dai is presented and renders correctly dark: renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": null,
        "fontSize": 25,
      },
      Array [
        Object {
          "color": "#fcfcfc",
        },
        undefined,
      ],
      Object {
        "fontFamily": "icomoon",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`<Icon /> icon coin-dai is presented and renders correctly light: renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": null,
        "fontSize": 25,
      },
      Array [
        Object {
          "color": "#4b4c4d",
        },
        undefined,
      ],
      Object {
        "fontFamily": "icomoon",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`<Icon /> icon coin-eth is presented and renders correctly dark: renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": null,
        "fontSize": 25,
      },
      Array [
        Object {
          "color": "#fcfcfc",
        },
        undefined,
      ],
      Object {
        "fontFamily": "icomoon",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`<Icon /> icon coin-eth is presented and renders correctly light: renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": null,
        "fontSize": 25,
      },
      Array [
        Object {
          "color": "#4b4c4d",
        },
        undefined,
      ],
      Object {
        "fontFamily": "icomoon",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`<Icon /> icon coin-zrx is presented and renders correctly dark: renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": null,
        "fontSize": 25,
      },
      Array [
        Object {
          "color": "#fcfcfc",
        },
        undefined,
      ],
      Object {
        "fontFamily": "icomoon",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`<Icon /> icon coin-zrx is presented and renders correctly light: renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": null,
        "fontSize": 25,
      },
      Array [
        Object {
          "color": "#4b4c4d",
        },
        undefined,
      ],
      Object {
        "fontFamily": "icomoon",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`<Icon /> icon contact is presented and renders correctly dark: renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": null,
        "fontSize": 25,
      },
      Array [
        Object {
          "color": "#fcfcfc",
        },
        undefined,
      ],
      Object {
        "fontFamily": "icomoon",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`<Icon /> icon contact is presented and renders correctly light: renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": null,
        "fontSize": 25,
      },
      Array [
        Object {
          "color": "#4b4c4d",
        },
        undefined,
      ],
      Object {
        "fontFamily": "icomoon",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`<Icon /> icon cross-arrows is presented and renders correctly dark: renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": null,
        "fontSize": 25,
      },
      Array [
        Object {
          "color": "#fcfcfc",
        },
        undefined,
      ],
      Object {
        "fontFamily": "icomoon",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`<Icon /> icon cross-arrows is presented and renders correctly light: renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": null,
        "fontSize": 25,
      },
      Array [
        Object {
          "color": "#4b4c4d",
        },
        undefined,
      ],
      Object {
        "fontFamily": "icomoon",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`<Icon /> icon edit is presented and renders correctly dark: renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": null,
        "fontSize": 25,
      },
      Array [
        Object {
          "color": "#fcfcfc",
        },
        undefined,
      ],
      Object {
        "fontFamily": "icomoon",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`<Icon /> icon edit is presented and renders correctly light: renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": null,
        "fontSize": 25,
      },
      Array [
        Object {
          "color": "#4b4c4d",
        },
        undefined,
      ],
      Object {
        "fontFamily": "icomoon",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`<Icon /> icon full-star is presented and renders correctly dark: renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": null,
        "fontSize": 25,
      },
      Array [
        Object {
          "color": "#fcfcfc",
        },
        undefined,
      ],
      Object {
        "fontFamily": "icomoon",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`<Icon /> icon full-star is presented and renders correctly light: renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": null,
        "fontSize": 25,
      },
      Array [
        Object {
          "color": "#4b4c4d",
        },
        undefined,
      ],
      Object {
        "fontFamily": "icomoon",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`<Icon /> icon gas-station is presented and renders correctly dark: renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": null,
        "fontSize": 25,
      },
      Array [
        Object {
          "color": "#fcfcfc",
        },
        undefined,
      ],
      Object {
        "fontFamily": "icomoon",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`<Icon /> icon gas-station is presented and renders correctly light: renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": null,
        "fontSize": 25,
      },
      Array [
        Object {
          "color": "#4b4c4d",
        },
        undefined,
      ],
      Object {
        "fontFamily": "icomoon",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`<Icon /> icon gas-station-eth is presented and renders correctly dark: renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": null,
        "fontSize": 25,
      },
      Array [
        Object {
          "color": "#fcfcfc",
        },
        undefined,
      ],
      Object {
        "fontFamily": "icomoon",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`<Icon /> icon gas-station-eth is presented and renders correctly light: renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": null,
        "fontSize": 25,
      },
      Array [
        Object {
          "color": "#4b4c4d",
        },
        undefined,
      ],
      Object {
        "fontFamily": "icomoon",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`<Icon /> icon info is presented and renders correctly dark: renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": null,
        "fontSize": 25,
      },
      Array [
        Object {
          "color": "#fcfcfc",
        },
        undefined,
      ],
      Object {
        "fontFamily": "icomoon",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`<Icon /> icon info is presented and renders correctly light: renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": null,
        "fontSize": 25,
      },
      Array [
        Object {
          "color": "#4b4c4d",
        },
        undefined,
      ],
      Object {
        "fontFamily": "icomoon",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`<Icon /> icon key is presented and renders correctly dark: renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": null,
        "fontSize": 25,
      },
      Array [
        Object {
          "color": "#fcfcfc",
        },
        undefined,
      ],
      Object {
        "fontFamily": "icomoon",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`<Icon /> icon key is presented and renders correctly light: renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": null,
        "fontSize": 25,
      },
      Array [
        Object {
          "color": "#4b4c4d",
        },
        undefined,
      ],
      Object {
        "fontFamily": "icomoon",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`<Icon /> icon menu-open is presented and renders correctly dark: renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": null,
        "fontSize": 25,
      },
      Array [
        Object {
          "color": "#fcfcfc",
        },
        undefined,
      ],
      Object {
        "fontFamily": "icomoon",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`<Icon /> icon menu-open is presented and renders correctly light: renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": null,
        "fontSize": 25,
      },
      Array [
        Object {
          "color": "#4b4c4d",
        },
        undefined,
      ],
      Object {
        "fontFamily": "icomoon",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`<Icon /> icon outline-star is presented and renders correctly dark: renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": null,
        "fontSize": 25,
      },
      Array [
        Object {
          "color": "#fcfcfc",
        },
        undefined,
      ],
      Object {
        "fontFamily": "icomoon",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`<Icon /> icon outline-star is presented and renders correctly light: renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": null,
        "fontSize": 25,
      },
      Array [
        Object {
          "color": "#4b4c4d",
        },
        undefined,
      ],
      Object {
        "fontFamily": "icomoon",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`<Icon /> icon password is presented and renders correctly dark: renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": null,
        "fontSize": 25,
      },
      Array [
        Object {
          "color": "#fcfcfc",
        },
        undefined,
      ],
      Object {
        "fontFamily": "icomoon",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`<Icon /> icon password is presented and renders correctly light: renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": null,
        "fontSize": 25,
      },
      Array [
        Object {
          "color": "#4b4c4d",
        },
        undefined,
      ],
      Object {
        "fontFamily": "icomoon",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`<Icon /> icon password-with-asterics is presented and renders correctly dark: renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": null,
        "fontSize": 25,
      },
      Array [
        Object {
          "color": "#fcfcfc",
        },
        undefined,
      ],
      Object {
        "fontFamily": "icomoon",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`<Icon /> icon password-with-asterics is presented and renders correctly light: renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": null,
        "fontSize": 25,
      },
      Array [
        Object {
          "color": "#4b4c4d",
        },
        undefined,
      ],
      Object {
        "fontFamily": "icomoon",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`<Icon /> icon right-arrow is presented and renders correctly dark: renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": null,
        "fontSize": 25,
      },
      Array [
        Object {
          "color": "#fcfcfc",
        },
        undefined,
      ],
      Object {
        "fontFamily": "icomoon",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`<Icon /> icon right-arrow is presented and renders correctly light: renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": null,
        "fontSize": 25,
      },
      Array [
        Object {
          "color": "#4b4c4d",
        },
        undefined,
      ],
      Object {
        "fontFamily": "icomoon",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`<Icon /> icon search is presented and renders correctly dark: renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": null,
        "fontSize": 25,
      },
      Array [
        Object {
          "color": "#fcfcfc",
        },
        undefined,
      ],
      Object {
        "fontFamily": "icomoon",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`<Icon /> icon search is presented and renders correctly light: renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": null,
        "fontSize": 25,
      },
      Array [
        Object {
          "color": "#4b4c4d",
        },
        undefined,
      ],
      Object {
        "fontFamily": "icomoon",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`<Icon /> icon selection-arrow is presented and renders correctly dark: renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": null,
        "fontSize": 25,
      },
      Array [
        Object {
          "color": "#fcfcfc",
        },
        undefined,
      ],
      Object {
        "fontFamily": "icomoon",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`<Icon /> icon selection-arrow is presented and renders correctly light: renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": null,
        "fontSize": 25,
      },
      Array [
        Object {
          "color": "#4b4c4d",
        },
        undefined,
      ],
      Object {
        "fontFamily": "icomoon",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`<Icon /> icon settings is presented and renders correctly dark: renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": null,
        "fontSize": 25,
      },
      Array [
        Object {
          "color": "#fcfcfc",
        },
        undefined,
      ],
      Object {
        "fontFamily": "icomoon",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`<Icon /> icon settings is presented and renders correctly light: renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": null,
        "fontSize": 25,
      },
      Array [
        Object {
          "color": "#4b4c4d",
        },
        undefined,
      ],
      Object {
        "fontFamily": "icomoon",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`<Icon /> icon trade is presented and renders correctly dark: renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": null,
        "fontSize": 25,
      },
      Array [
        Object {
          "color": "#fcfcfc",
        },
        undefined,
      ],
      Object {
        "fontFamily": "icomoon",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`<Icon /> icon trade is presented and renders correctly light: renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": null,
        "fontSize": 25,
      },
      Array [
        Object {
          "color": "#4b4c4d",
        },
        undefined,
      ],
      Object {
        "fontFamily": "icomoon",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`<Icon /> icon trash is presented and renders correctly dark: renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": null,
        "fontSize": 25,
      },
      Array [
        Object {
          "color": "#fcfcfc",
        },
        undefined,
      ],
      Object {
        "fontFamily": "icomoon",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`<Icon /> icon trash is presented and renders correctly light: renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": null,
        "fontSize": 25,
      },
      Array [
        Object {
          "color": "#4b4c4d",
        },
        undefined,
      ],
      Object {
        "fontFamily": "icomoon",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`<Icon /> icon wallet is presented and renders correctly dark: renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": null,
        "fontSize": 25,
      },
      Array [
        Object {
          "color": "#fcfcfc",
        },
        undefined,
      ],
      Object {
        "fontFamily": "icomoon",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`<Icon /> icon wallet is presented and renders correctly light: renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": null,
        "fontSize": 25,
      },
      Array [
        Object {
          "color": "#4b4c4d",
        },
        undefined,
      ],
      Object {
        "fontFamily": "icomoon",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`<Icon /> icon warning is presented and renders correctly dark: renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": null,
        "fontSize": 25,
      },
      Array [
        Object {
          "color": "#fcfcfc",
        },
        undefined,
      ],
      Object {
        "fontFamily": "icomoon",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`<Icon /> icon warning is presented and renders correctly light: renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": null,
        "fontSize": 25,
      },
      Array [
        Object {
          "color": "#4b4c4d",
        },
        undefined,
      ],
      Object {
        "fontFamily": "icomoon",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`<Icon /> light: renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": null,
        "fontSize": 25,
      },
      Array [
        Object {
          "color": "#4b4c4d",
        },
        undefined,
      ],
      Object {
        "fontFamily": "icomoon",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`<Icon /> name dark: renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": null,
        "fontSize": 25,
      },
      Array [
        Object {
          "color": "#fcfcfc",
        },
        undefined,
      ],
      Object {
        "fontFamily": "icomoon",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`<Icon /> name light: renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": null,
        "fontSize": 25,
      },
      Array [
        Object {
          "color": "#4b4c4d",
        },
        undefined,
      ],
      Object {
        "fontFamily": "icomoon",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`<Icon /> size dark: renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": null,
        "fontSize": 18,
      },
      Array [
        Object {
          "color": "#fcfcfc",
        },
        undefined,
      ],
      Object {
        "fontFamily": "icomoon",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`<Icon /> size light: renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": null,
        "fontSize": 18,
      },
      Array [
        Object {
          "color": "#4b4c4d",
        },
        undefined,
      ],
      Object {
        "fontFamily": "icomoon",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;
